decoder/build.rs

```diff
@@ -1,13 +1,31 @@
-use std::{env, error::Error, fs, path::PathBuf, process::Command};
+use std::{
+    env,
+    error::Error,
+    fs,
+    path::{Path, PathBuf},
+    process::Command,
+};
 
 use semver::Version;
 
 fn main() -> Result<(), Box<dyn Error>> {
     let out = &PathBuf::from(env::var("OUT_DIR")?);
-    let output = Command::new("git").args(&["rev-parse", "HEAD"]).output()?;
-    let version = if output.status.success() {
-        String::from_utf8(output.stdout).unwrap()
+    let hash = Command::new("git")
+        .args(&["rev-parse", "HEAD"])
+        .output()
+        .ok()
+        .and_then(|output| {
+            if output.status.success() {
+                String::from_utf8(output.stdout).ok()
+            } else {
+                None
+            }
+        });
+    let version = if let Some(hash) = hash {
+        hash
     } else {
+        assert!(!Path::new(".git").exists(), "you need to install the `git` command line tool to install the git version of `probe-run`");
```

Comment:
Did you test that this is the right path? I'm not sure in which working directory build scripts are executed. If it's the Cargo workspace root this seems correct, but it seems like it might be the *package* root, which means that this would be wrong for the decoder.

Reply:
yes, there's an old commit that removes a `panic!("{:?}, env::current_dir())` statement from the build script. I used to inspect the directory from which the build script was invoked. It was the package root and not the workspace root.

I guess I could also use `CARGO_MANIFEST_DIR` (was that the path to the directory that contains the `Cargo.toml` file?) instead of relying on `current_dir`. Perhaps that'd be more reliable?

Reply:
Either way seems fine to me